### ISSUE TITLE
Highlighted searched text not aligned properly

### ISSUE THREAD

**Issue by bhaskargoyal** [mozilla/pdf.js#7876]
Link to PDF file (or attach file here):  online [demo](https://mozilla.github.io/pdf.js/web/viewer.html)

Configuration:
- Web browser and its version: Google Chrome Version 54.0.2840.99 m (64-bit)
- Operating system and its version: Windows 10 Home Single Language
- PDF.js version: Used online demo

Steps to reproduce the problem:
1.  Open the demo site, as stated in README.md
2. Click on Search icon. 
3. Search "the".
4. Click on "Highlight all" option.
4. You may see the misalignment of highlighted text.

What is the expected behavior? (add screenshot)
Expected behavior should be to highlight the searched text without any misalignment.

What went wrong? (add screenshot)
![capture](https://cloud.githubusercontent.com/assets/12247314/20927946/af6b4710-bbe9-11e6-981a-749101556733.PNG)
###### Issue related to the sreenshot
You can see that as the x coordinate increases within the page, the selected text "the" and the highlighted one, gets further apart. As if it is a function of x coordinate.

**Comment by timvandermeij** [comment]
This has improved quite a bit after #7879, but it's not perfect yet.

**Comment by devanonyme** [comment]
I just have noticed the same problem while trying to figure out how to use search param.

You can reproduce it with the currently deployed demo (use 1.6.47?) : https://mozilla.github.io/pdf.js/web/viewer.html#search=Specialization

Most of occurances are ok but a few of them it looks like the hightlight is off by one character  (in Chrome and Firefox, it seems to be ok in Explorer).

**Comment by devanonyme** [comment]
Side note : 

This is not the case with the "Trace-based... " pdf sample but I have observed other hightlight shift (vertical by exemple) but these cases seems to be result of the OCR technology that was used to scan the source images of books (so really no much that can be done as it's impossible for the OCR to reproduce the font in all case).

In this sample, we can also see the shift visible in Adobe Acrobat Native (the ghosting of text in PDF.js is another issue, this occur when we use the browser text search instead of pdf.js text search) :

![capture_ecran 01-30-17 at 11 37 am](https://cloud.githubusercontent.com/assets/21067311/22431860/067daa08-e6e1-11e6-9425-706d6d7c50b0.PNG)

**Comment by urmary** [comment]
Hi,
Is anyone looking at this issue? This issue becomes more apparent if font is changed.

**Comment by xaviervansteene** [comment]
I had the same problem. I set 'text-spacing' to 'initial' of textLayer class in order to be sure text is not extended. It works well for me

**Comment by retrazil** [comment]
This issue still persists for online demo.

**Comment by timvandermeij** [comment]
Closing since this is fixed after recent search and text layer fixes.
